newsrevenuehub/rev-engine
# [NRH -5125] More DReason validation

#### What's this PR do?
Ensures proper conditional validation of "Reason for giving" options.

ReasonEditor was bypassing the requiredFields implementation.

#### How should this be manually tested?
In the editor, under "Reason for giving", make sure "Should filling this out be required?" is checked. Save the page, then try to donate without giving a reason.

#### Do any changes need to be made before deployment to staging? production? (adding environment variables, for example)?
No
